### PR TITLE
Alerting: Fix provisioning of contact points when contact point is renamed

### LINKED ISSUE #71619: Alerting: Provisioning does not correctly merge when renaming last receiver
### Description
When moving the last provisioned receiver in a contact point to another contact point, provisioning will create two contact points of the same name instead of merging them.
 
This should affect both terraform and file provisioning.

### Reproduction steps

1. Start a fresh Grafana instance and database.
2. Provision Grafana with the following alerting file:

```yaml
  - orgId: 1
    name: default contact
    receivers:
      - uid: b9bf06f8-bde2-4438-9d4a-bba0522dcd4d
        type: pagerduty
        settings:
          client: some client
          integrationKey: some key
          severity: criticalish
        disableResolveMessage: false
  - orgId: 1
    name: slack test
    receivers:
      - uid: cbfd0976-8228-4126-b672-4419f30a9e50
        type: slack
        settings:
          text: title body test
          title: title test
          url: some secure slack webhook
        disableResolveMessage: true
```
3. Move one receiver to the other contact point:
```yaml
  - orgId: 1
    name: default contact
    receivers:
      - uid: b9bf06f8-bde2-4438-9d4a-bba0522dcd4d
        type: pagerduty
        settings:
          client: some client
          integrationKey: some key
          severity: criticalish
        disableResolveMessage: false
      - uid: cbfd0976-8228-4126-b672-4419f30a9e50
        type: slack
        settings:
          text: title body test
          title: title test
          url: some secure slack webhook
        disableResolveMessage: true
```

Now there will be two contact points with the name `default contact`, one with each receiver.

### Expectation

1. There should be a single contact point with both receivers with the name `default contact`.
2. All references to `slack test` in notification policies should be renamed to `default contact` (Maybe not necessary?)

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This PR fixes a bug when the last contact point in the receiver group gets moved to another existing group. See the video below for a demonstration of the bug

<details><summary> Demonstartion of the bug</summary>

https://github.com/grafana/grafana/assets/25988953/e9c333c3-0389-4bc8-a03f-722a27972c11

</details>

**Why do we need this feature?**
This causes another bug - 2 receivers with the same name are not supported in Alertmanager. The last in the order wins, and the notification policy that before the update sent notifications to "discord", after the update sends __only__ to email.

**Who is this feature for?**
Those who use provisioning API

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/71619

**Special notes for your reviewer:**
The first commit introduces a unit test that demonstrates the bug.

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/provisioning/contactpoints.go [hunk 1]
```diff
@@ -491,6 +489,11 @@ groupLoop:
 						// Add the modified receiver to the new group...
 						candidateExistingGroup.GrafanaManagedReceivers = append(candidateExistingGroup.GrafanaManagedReceivers, target)
 						configModified = true
+
+						// if the receiver group turns out to be empty. Remove it.
```
Comment (alexweav) [inline pkg/services/ngalert/provisioning/contactpoints.go:493]:
```suggestion
						// if the old receiver group turns out to be empty. Remove it.
```
an incredibly tiny nitpick :smile:

**Review by alexweav — APPROVED** [review]
LGTM!

**Review by JacobsonMT — APPROVED** [review]
LGTM

**Comment by yuri-tceretian** [comment]
This is the duplicate of the PR https://github.com/grafana/grafana/pull/71620, which I reviewed and completely forgot about. Anyway, this PR has the same problem as I described in https://github.com/grafana/grafana/pull/71620#issuecomment-1648377482, i.e. it updates all routes that use the old receiver.

**Pull request merged**